Core/include/Acts/EventData/GenericParticleHypothesis.hpp

```diff
@@ -0,0 +1,87 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/PdgParticle.hpp"
+#include "Acts/EventData/Charge.hpp"
+#include "Acts/EventData/ChargeConcept.hpp"
+#include "Acts/Utilities/Concepts.hpp"
+
+#include <utility>
+
+namespace Acts {
+
+/// @brief Particle hypothesis used in reconstruction
+///
+/// Our reconstruction hypothesis consists of absolute PDG code, mass and
+/// absolute charge
+template <ACTS_CONCEPT(ChargeConcept) charge_t>
+class GenericParticleHypothesis {
```

Comment:
Should this go into `detail` to communicate that we want clients to use the types from `ParticleHypothesis.hpp`?

Reply:
shall we do this even tho we use public inheritance now?

Reply:
You mean if we publicly inherit, the base class should be outside of `detail`?

Reply:
Hmm I don't think that's necessary. Its quite usual in C++ to inherit from some base class that is not part of a public interface, even in boost and some std lib implementations I think...

Reply:
One interesting question would be if doxygen shows the inherited member function then however...